samcli/lib/config/samconfig.py

```diff
@@ -106,19 +115,26 @@ def put(self, cmd_names, section, key, value, env=DEFAULT_ENV):
 
         if not self.document:
             self._read()
-            # Empty document prepare the initial structure.
-            # self.document is a nested dict, we need to check each layer and add new tables, otherwise duplicated key
-            # in parent layer will override the whole child layer
-            if self.document.get(env, None):
-                if self.document[env].get(self._to_key(cmd_names), None):
-                    if not self.document[env][self._to_key(cmd_names)].get(section, None):
-                        self.document[env][self._to_key(cmd_names)].update({section: {key: value}})
+        # Empty document prepare the initial structure.
+        # self.document is a nested dict, we need to check each layer and add new tables, otherwise duplicated key
+        # in parent layer will override the whole child layer
+        cmd_name_key = self._to_key(cmd_names)
+        env_content = self.document.get(env, {})
+        cmd_content = env_content.get(cmd_name_key, {})
+        param_content = cmd_content.get(section, {})
+        if env_content:
+            if cmd_content:
+                if param_content:
+                    self.document[env][cmd_name_key][section].update({key: value})
```

Comment:
You can use `param_content` here.

Reply:
updated this one and the other two below.